### PR TITLE
Develop surface reaction

### PR DESCRIPTION
**Pull request by mattldawson**
Adds a surface reaction type for gas phase species:
```
X -> Y + Z + ...
```
where X, Y, and Z are gas-phase species and the reaction rate depends on total particle surface area for a given aerosol phase.

### CODE REVIEW

File: src/rxns/rxn_surface.F90 [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    do i_aero_rep = 1, size(aero_rep)
+      phase_ids = aero_rep(i_aero_rep)%val%phase_ids(phase_name)
+      n_aero_phase = n_aero_phase + size(phase_ids)
+      do i_phase = 1, size(phase_ids)
+        n_aero_jac_elem = n_aero_jac_elem + &
+            aero_rep(i_aero_rep)%val%num_jac_elem(phase_ids(i_phase))
+      end do
+    end do
+
+    allocate(this%condensed_data_int(NUM_INT_PROP_ + 2 + 3 * products%size() + &
```
Comment (K20shores) [inline src/rxns/rxn_surface.F90:201]: Could you leave a comment here and for the following allocate statement on what the constants are (+2 + 3 *)?

**Review by K20shores — APPROVED** [review]

File: src/util.h [hunk 2]
```diff
@@ -150,4 +161,58 @@ static inline double d_gas_aerosol_rxn_rate_constant_d_radius(
                           mean_free_path__m, radius__m, alpha));
 }
 
+/** Calculate the gas-aerosol reaction rate for the continuum regime
```
Comment (jcurtis2) [inline src/util.h:164]: Can we add some sort of citation for the source of this equation?

**Review by jcurtis2 — APPROVED** [review]

File: src/rxns/rxn_surface.F90 (hunk 1 above)
Comment (mattldawson) [reply src/rxns/rxn_surface.F90]: done

File: src/rxns/rxn_surface.c [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+        }
+      }
+    }
+  }
+  return;
+}
+
+/** \brief Update reaction data for new environmental conditions
+ *
+ * For surface reactions this only involves recalculating the rate
```
[the remark this answers is not in the document]
Comment (mattldawson) [reply src/rxns/rxn_surface.c]: updated the comment

File: src/rxns/rxn_surface.c [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+ */
+void rxn_surface_update_env_state(ModelData *model_data,
+                                            int *rxn_int_data,
+                                            double *rxn_float_data,
+                                            double *rxn_env_data) {
+  int *int_data = rxn_int_data;
+  double *float_data = rxn_float_data;
+  double *env_data = model_data->grid_cell_env;
+
+  // save the mean free path [m] for calculating condensation rates
```
[the remark this answers is not in the document]
Comment (mattldawson) [reply src/rxns/rxn_surface.c]: updated the comment

File: src/util.h (hunk 2 above)
Comment (mattldawson) [reply src/util.h]: added citation to Tie et al. 2003

**Review by jtgasparik — COMMENTED** [review]
none!

**Pull request merged**